lib/posix/posix_internal.h

```diff
@@ -36,16 +36,21 @@ struct posix_thread {
 	/* Exit status */
 	void *retval;
 
-	/* Pthread cancellation */
-	uint8_t cancel_state;
-	uint8_t cancel_type;
-	bool cancel_pending;
-
-	/* Detach state */
-	uint8_t detachstate;
-
-	/* Queue ID (internal-only) */
-	uint8_t qid;
+	union {
+		uint8_t: 8;
```

Comment:
😮

Reply:
🥷

Reply:
I should note, this is a kind of hackish way to only pack only a portion of a structure.

The issue with making the entire structure packed is that gcc generates a warning when taking the address of any member of a packed structure even (seemingly) when the offset of the member does not change if packed / unpacked.

Reply:
Are you sure `__packed` is necessary though? I did a quick compile test on my host machine without the `__packed` and `sizeof()` indicates a size of 1.

Reply:
Maybe not - that would be preferable.

Reply:
Dropped the last commit - due to padding, bitfields make no difference in the size of the structure.